runtime/interpreter/storage.go

```diff
@@ -33,31 +33,42 @@ import (
 	"github.com/onflow/cadence/runtime/common"
 )
 
-func StoredValue(storable atree.Storable, storage atree.SlabStorage) Value {
+func StoredValue(gauge common.MemoryGauge, storable atree.Storable, storage atree.SlabStorage) Value {
 	storedValue, err := storable.StoredValue(storage)
 	if err != nil {
 		panic(err)
 	}
 
-	return MustConvertStoredValue(storedValue)
+	return MustConvertStoredValue(gauge, storedValue)
 }
 
-func MustConvertStoredValue(value atree.Value) Value {
-	converted, err := ConvertStoredValue(value)
+func MustConvertStoredValue(gauge common.MemoryGauge, value atree.Value) Value {
+	converted, err := ConvertStoredValue(gauge, value)
 	if err != nil {
 		panic(err)
 	}
 	return converted
 }
 
-func ConvertStoredValue(value atree.Value) (Value, error) {
+func MustConvertUnmeteredStoredValue(value atree.Value) Value {
+	converted, err := ConvertStoredValue(nil, value)
+	if err != nil {
+		panic(err)
+	}
+	return converted
+}
+
+func ConvertStoredValue(gauge common.MemoryGauge, value atree.Value) (Value, error) {
```

Comment:
This PR adds metering for the array, dictionary, and composite cases in this function, but we don't do anything for any other values types. If values are being loaded from storage (hence the atree), do we need to "retroactively" meter them once they move from storage into the memory of the program?

Reply:
Its already handled by the decoder IINM
e.g:
https://github.com/onflow/cadence/blob/b5ebaced3696485167163b6dcd80335e2df2700d/runtime/interpreter/decode.go#L68-L81